modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/file_entry_history.jsp

```diff
@@ -27,7 +27,14 @@
 		status = WorkflowConstants.STATUS_ANY;
 	}
 
+	int fileVersionsLimit = 10;
+
+	int index = 0;
+
 	for (FileVersion fileVersion : fileEntry.getFileVersions(status)) {
+		if (FeatureFlagManagerUtil.isEnabled("LPS-175915") && (index >= fileVersionsLimit)) {
+			break;
+		}
```

Comment:
what do you think of doing something like this instead bring all the fileversions if there is a limit?

```
	int start = QueryUtil.ALL_POS;
	int end = QueryUtil.ALL_POS;

	if (FeatureFlagManagerUtil.isEnabled("LPS-175915") ) {
		start = 0;
		end = 10;
	}

	for (FileVersion fileVersion : fileEntry.getFileVersions(status, start, end)) {
```

Reply:
Makes sense, I've changed it, thanks!